docs/painless/painless-guide/painless-execute-script.asciidoc

```diff
@@ -1,37 +1,137 @@
+`POST /_scripts/painless/_execute`
+
+[[painless-execute-api-desc]]
+==== {api-description-title}
+Use this API to build and test scripts, such as when defining a script for a
+{ref}/runtime.html[runtime field]. This API requires very few dependencies, and is
+especially useful if you don't have permissions to write documents on a cluster.
+
+The API uses several _contexts_, which control how scripts are executed, what
+variables are available at runtime, and what the return type is.
+
+Each context requires a script, but additional parameters depend on the context
+you're using for that script.
+
+[[painless-execute-api-request-body]]
+==== {api-request-body-title}
+`script`:: (Required, object)
+The Painless script to execute.
++
+.Properties of `script`
+--
+include::../painless-contexts/painless-runtime-fields-context.asciidoc[tag=runtime-field-emit]
+--
+
+`context`:: (Optional, string)
+The context that the script should run in. Defaults to `painless_test` if no
+context is specified.
++
+.Properties of `context`
+[%collapsible%open]
+====
+`painless_test`::
+The default context if no other context is specified.
+
+`filter`::
+Treats scripts as if they were run inside a `script` query.
+
+`score`::
+Treats scripts as if they were run inside a `script_score` function in a
+`function_score` query.
 
-experimental[The painless execute api is new and the request / response format may change in a breaking way in the future]
-
-The Painless execute API allows an arbitrary script to be executed and a result to be returned.
-
-[[painless-execute-api-parameters]]
-.Parameters
-[options="header"]
-|======
-| Name              | Required  | Default                | Description
-| `script`          | yes       | -                      | The script to execute.
-| `context`         | no        | `painless_test`        | The context the script should be executed in.
-| `context_setup`   | no        | -                      | Additional parameters to the context.
-|======
-
-==== Contexts
-
-Contexts control how scripts are executed, what variables are available at runtime and what the return type is.
-
-===== Painless test context
-
-The `painless_test` context executes scripts as is and does not add any special parameters.
-The only variable that is available is `params`, which can be used to access user defined values.
-The result of the script is always converted to a string.
-If no context is specified then this context is used by default.
-
-*Example*
-
-Request:
+[[painless-execute-runtime-context]]
+.Runtime fields context
+[%collapsible%open]
+=====
+--
+The following contexts are specific to runtime fields.
+
+NOTE: Result ordering in the runtime field context is not guaranteed.
+--
+
+****
+`boolean_field`:: 
+Emits a `boolean` data type and returns a sorted list of `boolean` values. See
+<<painless-runtime-boolean,boolean_field>>.
+
+`date_field`:: 
+Emits a `long` data type and returns a sorted list of `long` values. See
```

Comment:
Maybe more like "The context used for `date` fields. `emit` takes a `long` and the script returns a sorted list of dates." or something.

Reply:
I think it'd be useful to have a link to the docs of the field. And to call out "this is exactly the same script as you send on the field".

Reply:
I updated the description for each of the field contexts and also added a link to the doc page for each related data type.